providers/Mora/mod.ts

```diff
@@ -0,0 +1,251 @@
+	}
+
+	override getLinkTypesForEntity(_entity: EntityId): LinkType[] {
+		return ['paid download'];
+	}
+
+	extractEmbeddedJson<Data>(webUrl: URL, maxTimestamp?: number): Promise<CacheEntry<WithApiUrl<Data>>> {
+		return this.fetchJSON<ApiArgs>(webUrl, {
+			policy: { maxTimestamp },
+			responseMutator: async (response) => {
+				const html = await response.text();
+				const metaTag = extractMetadataTag(html, 'msApplication-Arguments');
+				if (!metaTag) {
+					throw new ResponseError(this.name, 'Response is missing the expected <meta> tag', webUrl);
+				}
+
+				const apiArgsRaw = metaTag.replace(/&quot;/g, '"');
+				try {
+					const apiArgs: ApiArgs = JSON.parse(apiArgsRaw);
+
+					if (apiArgs) {
+						return new Response(JSON.stringify(apiArgs), response);
+					}
+				} catch (_error) {
+					throw new ResponseError(this.name, 'Failed to extract API arguments', webUrl);
+				}
+
+				throw new ResponseError(this.name, 'Failed to extract API arguments', webUrl);
+			},
+		}).then(({ content }) => {
+			const apiBase = apiUrl(content.mountPoint, content.labelId, content.materialNo);
+
+			const packageMetaUrl = new URL(apiBase);
+			packageMetaUrl.pathname += 'packageMeta.json';
+
+			return this.fetchJSON<WithApiUrl<Data>>(packageMetaUrl, {
+				policy: { maxTimestamp },
+				responseMutator: async (response) => {
+					const data = await response.json();
+					return new Response(
+						JSON.stringify({
+							apiUrl: apiBase,
+							data,
+						}),
+						response,
+					);
+				},
+			});
+		});
+	}
+}
+
+function apiUrl(mountPoint: string, labelId: string, materialNo: string) {
+	const paddedMaterialNo = materialNo.padStart(10, '0');
+	const slicedMaterialNo = `${paddedMaterialNo.slice(0, 4)}/${paddedMaterialNo.slice(4, 7)}/${
+		paddedMaterialNo.slice(7)
+	}`;
+
+	return new URL(`https://cf.mora.jp/contents/package/${mountPoint}/${labelId}/${slicedMaterialNo}/`);
+}
+
+export class MoraReleaseLookup extends ReleaseLookup<MoraProvider, PackageMeta> {
+	rawReleaseUrl: URL | undefined;
+	apiUrl: URL | undefined;
+
+	constructReleaseApiUrl(): URL | undefined {
+		return undefined;
+	}
+
+	async getRawRelease(): Promise<PackageMeta> {
+		if (this.lookup.method === 'gtin') {
+			throw new ProviderError(this.provider.name, 'GTIN lookups are not supported');
+		}
+
+		// Entity is already defined for ID/URL lookups.
+		const webUrl = this.provider.constructUrl(this.entity!);
+		this.rawReleaseUrl = webUrl;
+		const { content: release, timestamp } = await this.provider.extractEmbeddedJson<PackageMeta>(
+			webUrl,
+			this.options.snapshotMaxTimestamp,
+		);
+		this.apiUrl = release.apiUrl;
+		this.updateCacheTime(timestamp);
+
+		return release.data;
+	}
+
+	convertRawRelease(albumPage: PackageMeta): HarmonyRelease {
+		const label = { name: albumPage.labelcompanyname };
+		const tracklist = Object.entries(albumPage.trackList).map(([index, track]) =>
+			this.convertRawTrack(Number(index), track)
+		);
```

Comment:
```suggestion
		const tracklist = albumPage.trackList.map((track, index) =>
			this.convertRawTrack(index, track)
		);
```
would be the way to pass on the index, but since we don't need it, we can drop it from the methods parameters and simplify this further to
```suggestion
		const tracklist = albumPage.trackList.map(this.convertRawTrack.bind(this));
```